src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -1698,7 +1770,8 @@ rpmostree_compose_builtin_tree (int             argc,
       return FALSE;
     }
 
-  if (!opt_repo)
+  const char *rojig_outputdir = opt_ex_jigdo_output_rpm ?: opt_ex_jigdo_output_set;
+  if (!opt_repo && !rojig_outputdir)
```

Comment:
The various combinations of `--repo`, `--ex-rojig-output-rpm`, `--ex-rojig-output-set`, `--cachedir`, `--workdir`, `--unified-core` is hard to fully comprehend. I'm wondering if there are any ways we can split these flows apart now into distinct functions (or even files) across e.g. unified/non-unified, or ostree/rojig?

OK, and now I reread your older comment which mentions exactly this:

> What I find myself wondering though is if we really need to support the case of "compose both to ostree and rojig" here. If we didn't support all 3 cases, it might end up cleaner to refactor to separate C files with shared helpers.

Yes, I think that makes sense to me. The major case I can imagine where you want both ostree and rojig is to reduce friction when transitioning from one to the other. But as you mentioned, one can still use `commit2rojig`.

Reply:
OK so: split off a prep commit that drops the functionality, or do it in one go?

Reply:
Splitting off would be nice, though if it's too entangled, doing it in one go is fine.